Revise binary-search instructions (#2223)
exercism/problem-specifications: exercises/binary-search/instructions.md

A binary search algorithm finds an item in a list by repeatedly splitting it in half, only keeping the half which contains the item we're looking for.
It allows us to quickly narrow down the possible locations of our item until we find it, or until we've eliminated all possible locations.

> ````exercism/caution

Author: ```suggestion
~~~~exercism/caution
```

Binary search only works when a list has been sorted.

> ````

Reviewer: It's either three backticks, or four tildes, according to the styleguide. (See the section called **Special blocks** here). https://exercism.org/docs/building/markdown/markdown

Reviewer 2: That section says 4 tildes unless in exceptional circumstances, which this doesn't seem to be?

Author: Reading is very hard, okay.

Author: ```suggestion
~~~~
```

Reviewer: @SaschaMann I'm not sure this was clear, but I was agreeing with you 😄 

I've submitted a PR that fixes this in main, so once that is merged we can rebase this branch against that one.

The algorithm looks like this:

- Find the middle element of a sorted list and compare it with the item we're looking for.

> - If the middle element is our item, then we're done: return its index.

Reviewer: We need to avoid saying anything about the implementation here, because different tracks will tackle this differently.

Author: Makes sense!

```suggestion
- If the middle element is our item, then we're done!
```

- If the middle element is greater than our item, we can eliminate that element and all the elements **after** it.
- If the middle element is less than our item, we can eliminate that element and all the elements **before** it.

> - Repeat the process on the part of the list that has not been eliminated. If all the list has been eliminated then the item is not in the list.

Reviewer 3: Split this into two bullet points?

```suggestion
- If list entirely list has been eliminated then the item is not in the list.
- Repeat the process on the part of the list that has not been eliminated. 
```

Reviewer: Ah, I like that.

Author: How about this?

```suggestion
- If every element of the list has been eliminated then the item is not in the list.
- Otherwise, repeat the process on the part of the list that has not been eliminated.
```

Reviewer: Yeah, that's nice, @cmcaine. It  flows even better.
